Feat[home제작, 미디어쿼리,useEffect]

Kim-jaeyeon opened the pull request:
home페이지를 제작하였다. react-responsive 라이브러리를 활용하여 반응형을 제작하였고, useEffect를 통해 실시간 선택 배열 상태관리를 동작하도록 했다.  (페이지 전환은 아직 이루어지지 않은 상태)

Kim-jaeyeon committed:
Feat[home제작, 미디어쿼리,useEffect]

home페이지를 제작하였다. react-responsive 라이브러리를 활용하여 반응형을 제작하였고, useEffect를 통해 실시간 선택 배열 상태관리를 동작하도록 했다.  (페이지 전환은 아직 이루어지지 않은 상태)

jjh4450 commented on src/widgets/Footer.jsx:
```diff
@@ -3,7 +3,7 @@ import React from 'react'
   return (
     <>
-      <footer className="text-gray-600 body-font">
+      <footer className="bg-[#F5F5F5] text-gray-600 body-font">
```
배경 색상 별도로 설정하신 건가요?

jjh4450 replied:
전역적으로 사용된다면
`tailwind.config.js`에 색상을 추가하는게 더 좋을거 같아요
```js
/** @type {import('tailwindcss').Config} */
export default {
  content: [
    "./index.html",
    "./src/**/*.{js,ts,jsx,tsx}",
  ],
  theme: {
    extend: {
      fontFamily: {
        sans: ['"Pretendard Variable"', 'Pretendard', 'sans-serif'],
      },
      backgroundImage: {
        'kakao-login': "url('https://k.kakaocdn.net/14/dn/btroDszwNrM/I6efHub1SN5KCJqLm1Ovx1/o.jpg')",
      },
      colors:{
        'kakao-yellow': '#FFEB00',
        '변수명': '색상코드' //이 부분 편집
      }
    },
  },
  plugins: [],
}
```
`kakao-yellow` 사용 예시 첨부합니다!
```html
<div className="flex items-center bg-kakao-yellow">
```

jjh4450 commented on src/widgets/Header.jsx:
```diff
@@ -4,24 +4,8 @@ import { Link } from 'react-router-dom'
-            </svg>
-          </button>
-        </div>
+      <header className="text-gray-600 body-font h-8 bg-[#F5F5F5]">
```
위와 똑같은 의견 입니다!

jjh4450 commented on src/page/Home.jsx:
```diff
@@ -16,42 +17,23 @@ function Start() {
      * 선택된 알파벳을 추적하기 위한 상태
      * @type {Array}
      */
+    const isMobile = useMediaQuery({ query: '(max-width: 500px)' }); // 미디어 쿼리 라이브러리를 활용한 반응형 제작
```
tailwind css에서 자체적으로 미디어 쿼리를 [지원](https://tailwindcss.com/docs/responsive-design) 해주는데 굳이 필요한가 싶습니다
또한 추후 데스크탑도 강제적으로, 모바일로 보이게끔 하는 기능을 제작 중이라 개발자 모드에서 모바일을 사용해 주시면 충분할 거 같습니다

jjh4450 commented on index.html:
```diff
@@ -3,7 +3,7 @@
     <meta charset="UTF-8" />
     <link rel="icon" type="image/x-icon" href="/favicon.ico" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no" />
```
`user-scalable=no`를 지정한 이유가 궁금합니다!

Kim-jaeyeon committed:
Feat[전역 색상 설정, 페이지 이동 기초 설정, 디자인 다듬기]

전역 색상을 tailwind.config.js에 적고 적용/ react-router의 Link태그로 다음 페이지인 '/text' 로 넘어가도록 기초 설정/ 흰 버튼 테두리 색상등 디자인 다듬기
